src/framework/theme/components/select/select.component.ts

```diff
@@ -672,7 +670,19 @@ export class NbSelectComponent<T> implements OnInit, AfterViewInit, AfterContent
 
   show() {
     if (this.isHidden) {
+      if (!this.ref) {
+        this.createOverlay();
+        this.subscribeOnPositionChange();
+        this.createKeyManager();
+        this.subscribeOnOverlayKeys();
+      }
+
       this.ref.attach(this.portal);
+      if (this.selectionModel.length) {
```

Comment:
Let's not put *if* in *if*. Please, move in the separate function.

Reply:
Fixed c8822b2